### PR TITLE
feat: simplify branch protection by using a single check that affirms other jobs passed

### PR DESCRIPTION
**Pull request by onerandomusername**
## Checklist

- [x] this is a ci change

### CODE REVIEW

File: .github/workflows/lint-test.yml [hunk 1]
```diff
@@ -119,8 +119,9 @@ jobs:
         no-comments: true  # only add comments for one platform (see above)
         warnings: true
 
-
-  checks:
+  # these don't need to be ran independently, so we have one check that runs all of them
+  # needs a better name than this, too
+  small-jobs:
```
Comment (shiftinv) [inline .github/workflows/lint-test.yml:124]:
```suggestion
  misc:
```
mayhaps

File: .github/workflows/lint-test.yml [hunk 2]
```diff
@@ -258,3 +259,24 @@ jobs:
...
+    - docs
+    - pyright
+    - small-jobs
+    - pytest
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Decide whether the needed jobs succeeded or failed
+      uses: re-actors/alls-green@release/v1
```
Comment (shiftinv) [inline .github/workflows/lint-test.yml:280]:
```suggestion
      uses: re-actors/alls-green@v1.2.1
```

Comment (onerandomusername) [reply .github/workflows/lint-test.yml]: https://github.com/DisnakeDev/disnake/pull/790/commits/03c75744959847412c695c99857c5961c7dedd65

File: .github/workflows/lint-test.yml (hunk 1 above)
Comment (onerandomusername) [reply .github/workflows/lint-test.yml]: https://github.com/DisnakeDev/disnake/pull/790/commits/18b862bbf2739843f40ee5b6a03a53275c04066f

**Review by shiftinv — APPROVED** [review]
lgtm! this should simplify branch protection rules by a lot.

**Pull request merged**